sound/soc/sof/intel/hda-ipc.c

```diff
@@ -244,7 +244,8 @@ irqreturn_t hda_dsp_ipc_irq_handler(int irq, void *context)
 	int ret = IRQ_NONE;
 	u32 irq_status;
 
-	spin_lock(&sdev->hw_lock);
+	/* The function can be called at irq thread, so use spin_lock_irq */
+	spin_lock_irq(&sdev->hw_lock);
```

Comment:
question:
I can never remember the rules for spin_locks.

Do we need to protect with the _irq suffix only here, or also in the thread?

Reply:
I believe it has to be protected with _irq in the thread and not necessarily in the IRQ handler (you'd use _irq if you're not 100% certain the code runs only in IRQ handlers). IIRC, that is.

Reply:
Thanks @paulstelian97 for the explanation . @plbossart hda_dsp_ipc_irq_handler() will be called in the irq thread that's why I add the _irq suffix here. But actually I don't know if we do need the spin_lock or not. @RanderWang discussed it with me about it, but we didn't have a conclusion.